src/retractions.jl

```diff
@@ -56,6 +131,52 @@ matrix / matrices for point and tangent vector on a [`AbstractManifold`](@ref)
 """
 struct QRRetraction <: AbstractRetractionMethod end
 
+"""
+    SoftmaxRetraction <: AbstractRetractionMethod
+
+Describes a retraction that is based on the softmax function.
+"""
+struct SoftmaxRetraction <: AbstractRetractionMethod end
```

Comment:
Do we have an example for this retraction?

Reply:
I took all types that we have in Manifolds.jl and brought them here (export them here, use them there)m it is this one https://juliamanifolds.github.io/Manifolds.jl/latest/manifolds/probabilitysimplex.html#Manifolds.SoftmaxRetraction

Reply:
Introducing all types here means we also define all lower level 3 functions here, so the user does not have to do that thing on level 2 but only import them ad define the details. That's why I introduce them here.

Reply:
OK, thanks, that's fine :+1: